tests/phpunit/tests/block-template-utils.php

```diff
@@ -86,6 +86,28 @@ public function set_up() {
 		switch_theme( self::TEST_THEME );
 	}
 
+	/**
+	 * Tear down after each test.
+	 *
+	 * @since 6.5.0
+	 */
+	public function tear_down() {
+		global $wp_current_filter;
+
+		if (
+			'rest_pre_insert_wp_template' === current_filter() ||
+			'rest_pre_insert_wp_template_part' === current_filter()
+		) {
+			array_pop( $wp_current_filter );
```

Comment:
What's preventing calling `array_pop` inside the test before assertions start?

Reply:
I thought that we typically remove mocks in teardown functions, as they're run even if a test fails (same e.g. for block registration).

It's true that if we run it _before_ the first assertion in a test, it won't be a problem if the assertion fails; but there could still be a non-failure error.

Reply:
Both approaches work. I often see `add_filter` and `remove_filter` before assertions, but I don't have strong preferences.

Reply:
Some follow-up in #6244.